server/src/main/java/org/elasticsearch/search/dfs/DfsPhase.java

```diff
@@ -215,6 +215,13 @@ static DfsKnnResults singleKnnSearch(Query knnQuery, int k, Profilers profilers,
                 CollectorResult.REASON_SEARCH_TOP_HITS
             );
             topDocs = searcher.search(knnQuery, ipcm);
+
+            if (knnQuery instanceof ProfilingKnnFloatVectorQuery) {
```

Comment:
```suggestion
            if (knnQuery instanceof ProfilingKnnFloatVectorQuery profilingQuery) {
```

Then use `profilingQuery` and don't bother casting. 

Same for the `else if`